eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentHandler.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.payment
+
+import akka.actor.{Actor, ActorLogging, Props}
+import fr.acinq.eclair.NodeParams
+import fr.acinq.eclair.payment.handlers.{MultiPartHandler, ReceiveHandler}
+
+/**
+ * Generic payment handler that delegates handling of incoming messages to a list of handlers.
+ *
+ * @param nodeParams
+ */
+class PaymentHandler(nodeParams: NodeParams) extends Actor with ActorLogging {
+
+  override def receive: Receive = {
+    val defaultHandler = new MultiPartHandler(nodeParams, nodeParams.db.payments)
+    normal(Seq(defaultHandler), defaultHandler.handle(context, log))
```

Comment:
It looks like we're going to create a new  MultiPartHandler every time receive is called (I have not tested this)

Reply:
Addressed with 0f4d028.